src/device/pf_pdport.c

```diff
@@ -228,69 +299,117 @@ static void pf_pdport_remove_all_diag (pnet_t * net)
  */
 int pf_pdport_init (pnet_t * net)
 {
-   if (pf_pdport_load (net) != 0)
+   int port;
+   pf_port_iterator_t port_iterator;
+
+   pf_port_init_iterator_over_ports (net, &port_iterator);
+   port = pf_port_get_next (&port_iterator);
+   while (port != 0)
    {
-      /* Create file if missing */
-      (void)pf_pdport_save (net);
+      if (pf_pdport_load (net, port) != 0)
+      {
+         /* Create file if missing */
+         (void)pf_pdport_save (net, port);
+      }
+
+      port = pf_port_get_next (&port_iterator);
    }
    return 0;
 }
 
 int pf_pdport_reset_all (pnet_t * net)
 {
+   int port;
+   pf_port_iterator_t port_iterator;
+   pf_port_t * p_port_data = NULL;
    const char * p_file_directory = NULL;
-   pf_port_t * p_port_data = pf_port_get_state (net, PNET_PORT_1);
 
    (void)pf_cmina_get_file_directory (net, &p_file_directory);
-   pf_file_clear (p_file_directory, PF_FILENAME_PDPORT);
 
-   memset (&p_port_data->pdport, 0, sizeof (p_port_data->pdport));
-   pf_pdport_remove_all_diag (net);
+   pf_port_init_iterator_over_ports (net, &port_iterator);
+   port = pf_port_get_next (&port_iterator);
+   while (port != 0)
+   {
+      p_port_data = pf_port_get_state (net, port);
+      memset (&p_port_data->pdport, 0, sizeof (p_port_data->pdport));
+      pf_pdport_remove_all_diag (net, port);
+      pf_file_clear (p_file_directory, pf_pdport_get_filename (port));
 
+      port = pf_port_get_next (&port_iterator);
+   }
    return 0;
 }
 
+void pf_pdport_remove_data_files (const char * file_directory)
+{
+   int port;
+   for (port = PNET_PORT_1; port <= PNET_MAX_PORT; port++)
+   {
+      pf_file_clear (file_directory, pf_pdport_get_filename (port));
+   }
+}
+
 void pf_pdport_ar_connected (pnet_t * net, const pf_ar_t * p_ar)
```

Comment:
Maybe `pf_pdport_ar_connect_ind()` 
Now it sounds like `bool pf_pdport_is_ar_connected()`
Just a suggestion

Reply:
Changed to pf_pdport_ar_connect_ind()
Done